core/src/test/java/io/grpc/ContextTest.java

```diff
@@ -407,4 +644,61 @@ public void innerDeadlineCompletesBeforeOuter() throws Exception {
     assertTrue(base.cause() instanceof TimeoutException);
     assertNotSame(base.cause(), child.cause());
   }
+
+  @Test
+  public void cancellationCancelsScheduledTask() {
+    ScheduledExecutorService service = SharedResourceHolder.get(Context.SCHEDULER);
+    try {
+      ThreadPoolExecutor executor = (ThreadPoolExecutor) service;
+      //executor.setRemoveOnCancelPolicy(true);
```

Comment:
Done. It was done not quite 30 minutes ago. If you refresh you should see it as gone.